src/main/java/sirius/biz/storage/layer2/BasicBlobStorageSpace.java

```diff
@@ -1698,6 +1716,68 @@ private boolean tryCreateVariantAsync(B blob, String variantName) {
         }
     }
 
+    /**
+     * Tries to create the requested variant if the variant dos not exist already.
+     *
+     * @param blob        the blob for which the variant is to be created
+     * @param variantName the variant to generate
+     * @param retries     the number of retries left
+     * @return a future holding the conversion process
+     */
+    private Future tryCreateVariant(B blob, String variantName, int retries) {
+        if (retries == 0) {
+            Future future = new Future();
+            future.fail(new IllegalStateException(Strings.apply(
+                    "Failed to execute an optimistic locked update after %s retries",
+                    NUMBER_OF_ATTEMPTS_FOR_OPTIMISTIC_LOCKS)));
+            return future;
+        }
+
+        V variant = tryFetchVariant(blob, variantName);
+
+        if (variant != null && Strings.isFilled(variant.getPhysicalObjectKey())) {
+            return new Future().success();
+        }
+
+        if (conversionEnabled) {
+            if (variant == null) {
+                variant = createVariant(blob, variantName);
+            }
+
+            if (detectAndRemoveDuplicateVariant(variant, blob, variantName)) {
+                // An optimistic lock error occurred (another thread or node attempted the same). So we backup,
+                // wait a short and random amount of time and retry...
+                Wait.randomMillis(0, 150);
+                // the variant does already exist
+                return tryCreateVariant(blob, variantName, retries - 1);
+            } else {
+                return invokeConversionPipelineAsync(blob, variant);
+            }
```

Comment:
Sollte der Kommentar "the variant does already exist" nicht im else-Zweig stehen?

Reply:
nope. `detectAndRemoveDuplicateVariant` - Detects if the variant is still unique. Note that this is kind of a negative approach - if the method returns <tt>true</tt> a collision was detected and the given variant was removed. The <tt>false</tt> case is the expected one which verified that the variant is unique and good to go.

Reply:
I improved the comment